cirq/ops/eigen_gate.py

```diff
@@ -102,6 +102,8 @@ def __init__(
                 `cirq.rx(t)` uses a `global_shift` of -0.5, which is why
                 `cirq.unitary(cirq.rx(pi))` equals -iX instead of X.
         """
+        if not isinstance(exponent, (int, float, sympy.Basic)):
```

Comment:
I think we should allow `complex` as long as the imaginary part is 0. Instead of doing this check, do:

```
if isinstance(exponent, complex) and exponent.imag:
    raise ValueError(...)
```

For example, this will mean that things like `fractions.Fraction` will work. It's more in line with python's duck typing philosophy.